pkgs/applications/editors/sublime/4/common.nix

```diff
@@ -0,0 +1,157 @@
+{ buildVersion, aarch64sha256, x64sha256, dev ? false }:
+
+{ lib, fetchurl, stdenv, xorg, glib, libglvnd, glibcLocales, gtk3, cairo, pango, libredirect, makeWrapper, wrapGAppsHook
+, pkexecPath ? "/run/wrappers/bin/pkexec"
+, writeScript, common-updater-scripts, curl, jq, gnugrep
+, openssl, bzip2, bash, unzip, zip
+}:
+
+let
+  pname = "sublimetext4";
+  packageAttribute = "sublime4${stdenv.lib.optionalString dev "-dev"}";
+  binaries = [ "sublime_text" "plugin_host-3.3" "plugin_host-3.8" "crash_reporter" ];
+  primaryBinary = "sublime_text";
+  primaryBinaryAliases = [ "subl" "sublime" "sublime4" ];
+  downloadUrl = "https://download.sublimetext.com/sublime_text_build_${buildVersion}_${arch}.tar.xz";
+  versionUrl = "https://www.sublimetext.com/updates/4/dev_update_check";
+  versionFile = lib.toString ./packages.nix;
+  archSha256 = {
+    "aarch64-linux" = aarch64sha256;
+    "x86_64-linux" = x64sha256;
+  }.${stdenv.hostPlatform.system};
+  arch = {
+    "aarch64-linux" = "arm64";
+    "x86_64-linux" = "x64";
+  }.${stdenv.hostPlatform.system};
+
+  libPath = lib.makeLibraryPath [ xorg.libX11 glib libglvnd openssl gtk3 cairo pango ];
+  redirects = [ "/usr/bin/pkexec=${pkexecPath}" ];
```

Comment:
I kept the package updated in my pkgset:
https://github.com/jtojnar/nixfiles/blob/master/pkgs/sublime4/common.nix
It turned out that libredirect is more trouble than it is worth so I got rid of it:

https://github.com/jtojnar/nixfiles/commit/38c413abb6c24f8ad38d4cfc1f1e1994f8fc740d

https://github.com/sublimehq/sublime_text/issues/3502